sema4ai/src/sema4ai_code/vendored_deps/ls_protocols.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, Optional, TypedDict, Union
+
+
+class _DiagnosticSeverity(object):
+    Error = 1
+    Warning = 2
+    Information = 3
+    Hint = 4
+
+
+class _PositionTypedDict(TypedDict):
+    # Line position in a document (zero-based).
+    line: int
+
+    # Character offset on a line in a document (zero-based). Assuming that
+    # the line is represented as a string, the `character` value represents
+    # the gap between the `character` and `character + 1`.
+    #
+    # If the character value is greater than the line length it defaults back
+    # to the line length.
+    character: int
+
+
+class _RangeTypedDict(TypedDict):
+    start: _PositionTypedDict
+    end: _PositionTypedDict
+
+
+class _DiagnosticsTypedDict(TypedDict, total=False):
+    # The range at which the message applies.
+    range: _RangeTypedDict
+
+    # The diagnostic's severity. Can be omitted. If omitted it is up to the
+    # client to interpret diagnostics as error, warning, info or hint.
+    severity: Optional[int]  # DiagnosticSeverity
+
+    # The diagnostic's code, which might appear in the user interface.
+    code: Union[int, str]
+
+    # An optional property to describe the error code.
+    #
+    # @since 3.16.0
+    codeDescription: Any
+
+    # A human-readable string describing the source of this
+    # diagnostic, e.g. 'typescript' or 'super lint'.
+    source: Optional[str]
+
+    # The diagnostic's message.
+    message: str
+
+    # Additional metadata about the diagnostic.
+    #
+    # @since 3.15.0
+    tags: list  # DiagnosticTag[];
```

Comment:
Could we specify the type of item expected for this list?

Reply:
We don't use `tags` anywhere... `DiagnosticTag` could be modeled, but I guess it won't bring any benefits at this point (if at some point someone decides to use a tag in the diagnostic then that can be properly specified).